src/sentry/seer/explorer/tools.py

```diff
@@ -1111,3 +1112,183 @@ def get_trace_item_attributes(
+
+    return {"data": items}
+
+
+def get_metric_attributes_for_trace(
+    *,
+    org_id: int,
+    trace_id: str,
+    metric_name: str = "",
+    metric_name_case_sensitive: bool = True,
+    stats_period: str | None = None,
+    start: str | None = None,
+    end: str | None = None,
+    project_id: int | None = None,
+    project_slug: str | None = None,
+    sampling_mode: SAMPLING_MODES = "NORMAL",
+    limit: int | None = 50,
+) -> dict[str, Any] | None:
+
+    _validate_date_params(stats_period=stats_period, start=start, end=end)
+
+    try:
+        organization = Organization.objects.get(id=org_id)
+    except Organization.DoesNotExist:
+        logger.warning("Organization not found", extra={"org_id": org_id})
+        return None
+
+    projects = list(
+        Project.objects.filter(
+            organization=organization,
+            status=ObjectStatus.ACTIVE,
+            **({"id": project_id} if project_id else {}),
+            **({"slug": project_slug} if project_slug else {}),
+        )
+    )
+
+    # Look up full trace id if a short id is provided.
+    if len(trace_id) < 32:
+        full_trace_id = _get_full_trace_id(trace_id, organization, projects)
+        if not full_trace_id:
+            logger.warning(
+                "execute_trace_query: No full trace id found for short trace id",
+                extra={"org_id": org_id, "trace_id": trace_id},
+            )
+            return None
+    else:
+        full_trace_id = trace_id
+
+    # Build the GetTraceRequest.
+    snuba_params = SnubaParams(
+        start=datetime.fromisoformat(start) if start else None,
+        end=datetime.fromisoformat(end) if end else None,
+        stats_period=stats_period,
+        projects=projects,
+        organization=organization,
+        sampling_mode=sampling_mode,
+    )
+
+    resolver = TraceMetrics.get_resolver(params=snuba_params, config=SearchResolverConfig())
+    meta = resolver.resolve_meta(referrer=Referrer.SEER_RPC, sampling_mode=sampling_mode)
+    request = GetTraceRequest(
+        meta=meta,
+        trace_id=full_trace_id,
+        items=[
+            GetTraceRequest.TraceItem(
+                item_type=meta.trace_item_type,
+                attributes=None,  # Returns all attributes.
+            )
+        ],
+    )
+    if limit and not metric_name:
+        request.limit = limit
+
+    # Query EAP EndpointGetTrace
+    response = get_trace_rpc(request)
+
+    items: list[dict[str, Any]]
+    for item_group in response.item_groups:
+        items = parse_get_trace_rpc_response(item_group, resolver)
+        break  # Should only be one item group for metrics
+
+    if metric_name:
+        items = (
+            [item for item in items if metric_name == item["metric.name"]]
+            if metric_name_case_sensitive
+            else [item for item in items if metric_name.lower() == item["metric.name"].lower()]
+        )
+
+    if limit:
+        items = items[:limit]
+
+    return {"data": items}
```

Comment:
the metrics and logs methods look so similar i wonder if they can be combined more